packages/browser/src/index.ts

```diff
@@ -335,21 +337,39 @@ export default class LogtoClient {
   }
 
   private async getOidcConfig(): Promise<OidcConfigResponse> {
-    if (!this.oidcConfig) {
-      const { endpoint } = this.logtoConfig;
-      const discoveryEndpoint = getDiscoveryEndpoint(endpoint);
-      this.oidcConfig = await fetchOidcConfig(discoveryEndpoint, this.requester);
+    if (this.oidcConfig) {
+      return this.oidcConfig;
     }
 
+    if (this.oidcConfigPromise) {
+      return this.oidcConfigPromise;
+    }
+
+    const { endpoint } = this.logtoConfig;
+    const discoveryEndpoint = getDiscoveryEndpoint(endpoint);
+    this.oidcConfigPromise = fetchOidcConfig(discoveryEndpoint, this.requester);
+    this.oidcConfig = await this.oidcConfigPromise;
+
     return this.oidcConfig;
   }
 
+  private async getJwtVerifyGetKey(): Promise<JWTVerifyGetKey> {
+    const { jwksUri } = await this.getOidcConfig();
```

Comment:
The code here is just for testing.

IF
```ts
const { jwksUri } = await this.getOidcConfig();
```
is not moved above
```ts
if (!this.jwtVerifyGetKey) {
```
`createRemoteJWKSet` will be called 2 times.

---

It's better to rewrite `getJwtVerifyGetKey` like `getOidcConfig` above.